### PR TITLE
solution

### PR DESCRIPTION
**Pull request by snuber-w**

### CODE REVIEW

File: src/figuresClasses.ts [hunk 1]
```diff
@@ -1,19 +1,99 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
+const error = new Error('somethig went wrong');
```
Comment (naziamoff) [inline src/figuresClasses.ts:4]: no, you want your errors to be maximally descriptive. I as a user comes here - passed irrelevant input. I don't know what I did wrong. But I should know

File: src/figuresClasses.ts [hunk 2]
```diff
@@ -1,19 +1,99 @@
...
-export class Circle {
+  isValidTriangle(): boolean {
+    const { a, b, c } = this;
 
+    return a > 0
+      && b > 0
+      && c > 0
+      && a < b + c
+      && b < a + c
+      && c < a + b;
```
Comment (naziamoff) [inline src/figuresClasses.ts:34]: should be separated into two different error messages

**Review by naziamoff — CHANGES_REQUESTED** [review]
good job 🔥 
Let's fix the errors issue and we're good

**Review by 33Orange — APPROVED** [review]
Well done !

**Pull request closed without merging**